packages/web/src/components/track/DownloadSection.tsx

```diff
@@ -72,14 +73,19 @@ export const DownloadSection = ({ trackId }: DownloadSectionProps) => {
     shallowEqual
   )
   const { stemTracks } = useCurrentStems({ trackId })
-  const shouldDisplayDownloadAll = stemTracks.length > 1
+  const { hasDownloadAccess } = useGatedContentAccess(track)
   const {
     price,
     shouldDisplayPremiumDownloadLocked,
     shouldDisplayPremiumDownloadUnlocked,
     shouldDisplayDownloadFollowGated,
     shouldDisplayOwnerPremiumDownloads
   } = useDownloadableContentAccess({ trackId })
+
+  // Hide the download all button if there aren't multiple downloads and if the user
+  // happens to not have stream access to the track
+  const shouldDisplayDownloadAll = stemTracks.length > 1 && hasDownloadAccess
```

Comment:
what if stemTracks.length == 1 and hasDownloadAccess. that's still 2 tracks and should show Download All

Reply:
Oh yeah.... this is interesting.

It should be 

downloadCount + stemCount > 1